augur/io.py

```diff
@@ -92,11 +93,15 @@ def read_metadata(metadata_file, id_columns=("strain", "name"), chunk_size=None)
         kwargs["chunksize"] = chunk_size
 
     # Inspect the first chunk of the metadata, to find any valid index columns.
-    metadata = pd.read_csv(
-        metadata_file,
-        iterator=True,
-        **kwargs,
-    )
+    try:
+        metadata = pd.read_csv(
+            metadata_file,
+            iterator=True,
+            **kwargs,
+        )
+    except FileNotFoundError:
+        print_err(f"ERROR: File '{metadata_file}' does not exist.")
+        sys.exit(1)
```

Comment:
Sorry if I'm jumping in too soon while you're working on this, @victorlin, but I would not catch the `FileNotFoundError` here. This relates to @tsibley's comment about moving away from `sys.exit` in our internals, where we expect the `io` functions to act like a Python library by raising exceptions.

Instead, we can catch this exception in the top-level commands like `augur filter` that call `read_metadata` and other functions. If some exceptions are common enough to apply to all subcommands (like the recursion error), we could promote handling of those exceptions to the try/except block in `__init__.py`.

Reply:
Ah ok that makes sense! I'll update.

Reply:
Done: e15852458eedace4ebae7a27bd8fd3d12b1795fe